src/Search/TypesenseTokenHandler.php

```diff
@@ -0,0 +1,46 @@
+<?php
+
+namespace App\Search;
+
+use ACSEO\TypesenseBundle\Finder\TypesenseQuery;
+use App\Entity\User;
+use Symfony\Bundle\SecurityBundle\Security;
+
+class TypesenseTokenHandler
+{
+    public function __construct(private readonly Security $security)
+    {
+    }
+
+    public function handle(array $tokens): TypesenseQuery
+    {
+        $query = new TypesenseQuery($tokens['TEXT'] ?? '', 'title,serie,extension,authors,tags,summary');
+
+        $filterstring = [];
+        $orderString = [];
+        foreach ($tokens as $class => $token) {
+            if ($class !== 'TEXT') {
+                $handler = new $class();
```

Comment:
I would instantiate the class after the "instance of" checks, and pass the user in the constructor.